core/cmd/shell.go

```diff
@@ -61,20 +62,48 @@ var (
 	initGlobalsOnce sync.Once
 	prometheus      *ginprom.Prometheus
 	grpcOpts        loop.GRPCOpts
+	beholderClient  beholder.OtelClient
 )
 
-func initGlobals(cfgProm config.Prometheus, cfgTracing config.Tracing, logger logger.Logger) error {
+func initGlobals(cfgProm config.Prometheus, cfgTracing config.Tracing, cfgBeholder config.Beholder, lggr logger.Logger) error {
 	// Avoid double initializations, but does not prevent relay methods from being called multiple times.
 	var err error
 	initGlobalsOnce.Do(func() {
 		prometheus = ginprom.New(ginprom.Namespace("service"), ginprom.Token(cfgProm.AuthToken()))
 		grpcOpts = loop.NewGRPCOpts(nil) // default prometheus.Registerer
-		err = loop.SetupTracing(loop.TracingConfig{
+
+		tracingCfg := loop.TracingConfig{
 			Enabled:         cfgTracing.Enabled(),
 			CollectorTarget: cfgTracing.CollectorTarget(),
 			NodeAttributes:  cfgTracing.Attributes(),
 			SamplingRatio:   cfgTracing.SamplingRatio(),
-			OnDialError:     func(error) { logger.Errorw("Failed to dial", "err", err) },
+			OnDialError:     func(error) { lggr.Errorw("Failed to dial", "err", err) },
+		}
+		if cfgBeholder.OtelExporterGRPCEndpoint() == "" {
+			beholderClient = beholder.NewNoopClient()
```

Comment:
just to clarify - from now on, we always expect a global beholder client to exist and have been initialized before we reach relayer/product code?

Reply:
Yes but the global field is just a quirk and it should actually be injected around from the application field. But in general it is saner to have no-op implementations instead of nil/enabled checks at callsites.